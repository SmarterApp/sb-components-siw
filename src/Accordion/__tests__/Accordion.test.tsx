import * as React from "react";
import { shallow } from "enzyme";
import { Accordion, AccordionProps } from "@src/index";

const accordionContent = (
  <div id="test-content">
    test<p>I'm open</p>
  </div>
);

const accordionProps: AccordionProps = {
  accordionTitle: "testAccordion",
  isOpen: false,
  toggleExpand: jest.fn()
};

describe("Accordion", () => {
  const wrapper = shallow(
    <Accordion {...accordionProps}>{accordionContent}</Accordion>
  );
  const wrapperOpen = shallow(
    <Accordion {...accordionProps} isOpen={true}>
      {accordionContent}
    </Accordion>
  );

  it("matches closed snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has no content when closed", () => {
    const content = wrapper.find("#test-content");
    expect(content).toHaveLength(0);
  });

  it("matches open snapshot", () => {
    expect(wrapperOpen).toMatchSnapshot();
  });

  it("has content when opened", () => {
    const content = wrapperOpen.find("#test-content");
    expect(content).toHaveLength(1);
  });

  it("calls handle show content when closed", () => {
    const button = wrapper.find(".accordion-bar");
    button.simulate("click");
    expect(accordionProps.toggleExpand).toHaveBeenCalled();
  });

  it("call handle show content when open", () => {
    const button = wrapperOpen.find(".accordion-bar");
    button.simulate("click");
    expect(accordionProps.toggleExpand).toHaveBeenCalled();
  });

  it("renders closed then open", () => {
    wrapper.setProps({ isOpen: true });
    const content = wrapper.find("#test-content");
    expect(content).toHaveLength(1);
  });
});
