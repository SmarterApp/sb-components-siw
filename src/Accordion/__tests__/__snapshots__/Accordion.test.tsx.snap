// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion matches closed snapshot 1`] = `
<div
  className="accordion-container"
>
  <div
    className="accordion-bar"
    onClick={[MockFunction]}
    role="button"
  >
    <div
      className="accordion-content"
    >
      testAccordion
      <span
        aria-hidden="true"
        className="fa fa-chevron-down carat"
      />
    </div>
  </div>
</div>
`;

exports[`Accordion matches open snapshot 1`] = `
<div
  className="accordion-container"
>
  <div
    className="accordion-bar"
    onClick={[MockFunction]}
    role="button"
  >
    <div
      className="accordion-content"
    >
      testAccordion
      <span
        aria-hidden="true"
        className="fa fa-chevron-up carat"
      />
    </div>
  </div>
  <div
    className="accordion-content"
  >
    <div
      id="test-content"
    >
      test
      <p>
        I'm open
      </p>
    </div>
  </div>
</div>
`;
